bld/build.sh

```diff
@@ -124,79 +123,79 @@ process_args ()
 # -----------------------------------------------------------------------------
 native_build()
 {
-	if [ $build_pack_only == 0 ]; then
-		echo -e "\033[1mBuilding native...\033[0m"
-		for c in ${build_configs[@]}; do
-			echo -e "\033[1m    ${c}...\033[0m"
-			mkdir -p "${build_root}/cmake/${c}"
-			cd "${build_root}/cmake/${c}" > /dev/null
-
-			cmake $toolset -DCMAKE_BUILD_TYPE=$c -Drun_unittests:BOOL=$run_unittests \
-			      -Dmem_check:BOOL=$with_memcheck -Duse_zlog:BOOL=$use_zlog "$repo_root" \
+    if [ $build_pack_only == 0 ]; then
+        echo -e "\033[1mBuilding native...\033[0m"
+        for c in ${build_configs[@]}; do
+            echo -e "\033[1m    ${c}...\033[0m"
+            mkdir -p "${build_root}/cmake/${c}"
+            cd "${build_root}/cmake/${c}" > /dev/null
+
+            cmake $toolset -DCMAKE_BUILD_TYPE=$c -Drun_unittests:BOOL=$run_unittests \
+                  -Dmem_check:BOOL=$with_memcheck -Duse_zlog:BOOL=$use_zlog "$repo_root" \
                               -DLWS_IPV6:BOOL=ON || \
-			    return 1
-			# Start as much parallel jobs as requested by the user.
-			# Until the load average equals the number of cores.
-			# Be verbose if something goes wrong
-			make -j$MAKE_PARALLEL_JOBS --load-average=`nproc` || make VERBOSE=1 || \
-				return 1
-			if [ $run_unittests == OFF ]; then
-				echo "Skipping unittests..."
-			else
-				#use doctored (-DPURIFY no-asm) openssl
-				export LD_LIBRARY_PATH=/usr/local/ssl/lib
-				ctest -C "$c" --output-on-failure || \
-					return 1
-				export LD_LIBRARY_PATH=
-			fi
-		done
-	fi
-	return 0
+                return 1
+            # Start as much parallel jobs as requested by the user.
+            # Until the load average equals the number of cores.
+            # Be verbose if something goes wrong
+            make -j$MAKE_PARALLEL_JOBS --load-average=`nproc` || make VERBOSE=1 || \
+                return 1
+            if [ $run_unittests == OFF ]; then
+                echo "Skipping unittests..."
+            else
+                #use doctored (-DPURIFY no-asm) openssl
+                export LD_LIBRARY_PATH=/usr/local/ssl/lib
```

Comment:
You could set the environment variable LD_LIBRARY_PATH just for the invocation of ctest as follows:

`LD_LIBRARY_PATH=/usr/local/ssl/lib ctest -C "$c" --output-on-failure || return 1`

So you could keep the envrionment variable untouched for other programs called by the script.